### PR TITLE
Spec validation option

### PR DESCRIPTION
**Pull request by RealHarshThakur**
Signed-off-by: Harsh Thakur <harshthakur9030@gmail.com>

Fixes: https://github.com/fission/fission/issues/1694

Validation check is made optional which gives around 2x performance boost.

---
This change is [Reviewable](https://reviewable.io/reviews/fission/fission/1865)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,5 +1,7 @@
 module github.com/fission/fission
 
+go 1.15
```
Comment (vishal-biyani) [inline go.mod:3]: Can you please remove this? I think we have some dependency on earlier versions

**Review by vishal-biyani — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (harshthakur9030) [reply go.mod]: Done!

File: pkg/fission-cli/flag/flag.go [hunk 2]
```diff
@@ -176,14 +176,15 @@ var (
...
-	SpecDry      = Flag{Type: Bool, Name: flagkey.SpecDry, Usage: "View the generated specs"}
+	SpecSave       = Flag{Type: Bool, Name: flagkey.SpecSave, Usage: "Save to the spec directory instead of creating on cluster"}
+	SpecDir        = Flag{Type: String, Name: flagkey.SpecDir, Usage: "Directory to store specs, defaults to ./specs"}
+	SpecName       = Flag{Type: String, Name: flagkey.SpecName, Usage: "Name for the app, applied to resources as a Kubernetes annotation"}
+	SpecDeployID   = Flag{Type: String, Name: flagkey.SpecDeployID, Aliases: []string{"id"}, Usage: "Deployment ID for the spec deployment config"}
+	SpecWait       = Flag{Type: Bool, Name: flagkey.SpecWait, Usage: "Wait for package builds"}
+	SpecWatch      = Flag{Type: Bool, Name: flagkey.SpecWatch, Usage: "Watch local files for change, and re-apply specs as necessary"}
+	SpecDelete     = Flag{Type: Bool, Name: flagkey.SpecDelete, Usage: "Allow apply to delete resources that no longer exist in the specification"}
+	SpecDry        = Flag{Type: Bool, Name: flagkey.SpecDry, Usage: "View the generated specs"}
+	SpecValidation = Flag{Type: String, Name: flagkey.SpecValidate, Usage: "Turns validations on/off"}
```
Comment (vishal-biyani) [inline pkg/fission-cli/flag/flag.go:187]:
```suggestion
	SpecValidation = Flag{Type: String, Name: flagkey.SpecValidate, Usage: "Turns server side validations of Fission objects on/off"}
```

**Review by vishal-biyani — APPROVED** [review]
Except for one minor change, looks good, thanks @RealHarshThakur

File: pkg/fission-cli/flag/flag.go (hunk 2 above)
Comment (RealHarshThakur) [reply pkg/fission-cli/flag/flag.go]: Done

**Pull request merged**